library/src/main/java/com/novoda/downloadmanager/LiteBatchFileBuilder.java

```diff
@@ -40,20 +40,61 @@ public BatchFileBuilder saveTo(String path) {
 
     @Override
     public BatchFileBuilder saveTo(String path, String fileName) {
-        this.path = Optional.fromNullable(path).filterNot("/"::equals);
+        this.path = Optional.fromNullable(path);
         this.fileName = Optional.fromNullable(fileName);
         return this;
     }
 
     @Override
     public BatchBuilder apply() {
-        String absolutePath = storageRoot.path() + File.separator
-                + downloadBatchId.rawId() + File.separator
-                + path.map((it) -> it + File.separator).or("")
-                + fileName.getOrElse(() -> FileNameExtractor.extractFrom(networkAddress));
+        String absolutePath = buildPath(
+                storageRoot.path(),
+                downloadBatchId.rawId(),
+                path.or(""),
+                fileName.getOrElse(() -> FileNameExtractor.extractFrom(networkAddress))
+        );
 
         parentBuilder.withFile(new BatchFile(networkAddress, downloadFileId, absolutePath));
         return parentBuilder;
     }
 
+    private String buildPath(String... paths) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        String storagePath = paths[0];
+        if (storagePath.startsWith(File.separator)) {
+            stringBuilder.append(File.separator);
+        }
+
+        for (String path : paths) {
+            if (path.isEmpty()) {
+                continue; // ignore empty paths
```

Comment:
Didn't remove the comment 👅

Reply:
Oh, i thought the comment was for the `storagePath`!